server/helpers/auth/google.coffee

```diff
@@ -0,0 +1,94 @@
+googleapis  = require 'googleapis'
+config      = require 'config'
+
+oauth2 = googleapis.oauth2('v2')
+calendar = googleapis.calendar('v3')
+
+googleAuth =
+
+  authenticate: (authCode, clientId, redirect_uri, callback) ->
+    oauth2Client = buildAuthClient clientId, redirect_uri
+    getAuthToken authCode, oauth2Client, (err, tokens) ->
+      oauth2Client.setCredentials(tokens)
+      if callback
+        callback err, oauth2Client, tokens
+
+  getUserInfo: (oauth2Client, callback) ->
+    oauth2.userinfo.get {
+      auth: oauth2Client
+    }, (err, googleUser) ->
+      if err
+        console.log "Googleapis User Info Error:", err
+      if callback
+        callback err, googleUser
+
+  getCalendarEventsList: (oauth2Client, callback) ->
+    calendar.events.list {
+      calendarId: 'primary'
+      auth: oauth2Client
+    }, (err, events)->
+      if err
+        console.log "Googleapis Calendar Events Error:", err
+      if callback
+        callback err, events
+
+  getCalendarFreeBusy: (oauth2Client, id, callback) ->
+    today = new Date()
+    weekFromToday = new Date(today.getTime() + 7 * 24 * 60 * 60 * 1000)
+    console.log weekFromToday.toString()
+    calendar.freebusy.query {
+      resource:
+        timeMin: today.toISOString()
+        timeMax: weekFromToday.toISOString()
+        items: [{
+          id: id
+        }]
+      auth: oauth2Client
+    }, (err, events)->
+      if err
+        console.log "Googleapis Calendar Events Error:", err
+      if callback
+        callback err, events
+
+  getCalendarsFromusers: (userList, eventsList, currentUser, callback) ->
+    getUsersCalendars userList, eventsList, currentUser, callback
+
+  getAuthClient: (clientId, redirectUri) ->
+    return buildAuthClient clientId, redirectUri
+
+# Private Methods
+getStoredAuthClient = (user) ->
+  clientId = config.googleAuthConfig.clientId
+  redirectUri = config.googleAuthConfig.redirectUri
+  oauth2Client = googleAuth.getAuthClient clientId, redirectUri
+  oauth2Client.setCredentials user.auth
+  return oauth2Client
+
+getAuthToken = (authCode, oauth2Client, callback)->
+  oauth2Client.getToken authCode, (err, tokens)->
+    if err
+      console.log "Googleapis Token Error:", err
+    if callback
+      return callback err, tokens
+
+buildAuthClient = (clientId, redirectUri)->
+  secret = config.googleAuthConfig.clientSecret
+  OAuth2 = googleapis.auth.OAuth2
+  oauth2Client = new OAuth2 clientId, secret, redirectUri
+  return oauth2Client
+
+getUsersCalendars = (userList, eventsList, currentUser, callback) ->
```

Comment:
Yes, I want to refactor everything so it's not a bunch of nested callbacks

Cheers,

Brandon Mikulka
Startups2Students Coordinator
http://startups2students.hackcu.org/

On Mar 3, 2016, 10:17 PM -0700, Alex Campbell<notifications@github.com>, wrote:
>  
> Inserver/helpers/auth/google.coffee(https://github.com/BeTandem/scheduler_backend/pull/52#discussion_r54991172):
>  
> >+ return oauth2Client>+>+getAuthToken = (authCode, oauth2Client, callback)->>+ oauth2Client.getToken authCode, (err, tokens)->>+  if err>+   console.log "Googleapis Token Error:", err>+  if callback>+   return callback err, tokens>+>+buildAuthClient = (clientId, redirectUri)->>+ secret = config.googleAuthConfig.clientSecret>+ OAuth2 = googleapis.auth.OAuth2>+ oauth2Client = new OAuth2 clientId, secret, redirectUri>+ return oauth2Client>+>+getUsersCalendars = (userList, eventsList, currentUser, callback) ->
>  
> Do we wanna do a promise here?
>  
>  
> —
> Reply to this email directly orview it on GitHub(https://github.com/BeTandem/scheduler_backend/pull/52/files#r54991172).
>  
>  
>